br/pkg/storage/locking.go

```diff
@@ -3,23 +3,130 @@
 package storage
 
 import (
+	"bytes"
 	"context"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
+	"math/rand"
 	"os"
+	"path"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb/br/pkg/logutil"
+	"go.uber.org/multierr"
 	"go.uber.org/zap"
 )
 
+// ExclusiveWrite is a write that in a strong consistency storage.
+//
+// It is pretty like a "write if not exist", but it is atomic:
+//
+// if the write is success and the file wasn't deleted, no other `ExclusiveWrite`
+// over the same file was success.
+//
+// For more details, check docs/design/2024-10-11-put-and-verify-transactions-for-external-storages.md.
+type ExclusiveWrite struct {
```

Comment:
I feel like `CondtionalWrite` or `ConditionalPut` is more descriptive. And `VerifyWriteContext ` could be `ConditionCheck` or `Precondtion`, something like that. What do you think?